dev/service-registry/devmode-scaladsl/src/main/scala/com/lightbend/lagom/internal/scaladsl/registry/ServiceRegistration.scala

```diff
@@ -6,16 +6,21 @@ package com.lightbend.lagom.internal.scaladsl.registry
 
 import java.net.URI
 
+import akka.Done
+import akka.actor.CoordinatedShutdown
 import com.lightbend.lagom.scaladsl.api.ServiceInfo
 import com.typesafe.config.Config
-import play.api.inject.ApplicationLifecycle
 import play.api.Logger
 
 import scala.concurrent.{ ExecutionContext, Future }
 import scala.util.{ Failure, Success }
 
-class ServiceRegistration(serviceInfo: ServiceInfo, lifecycle: ApplicationLifecycle, config: Config,
-                          registry: ServiceRegistry)(implicit ec: ExecutionContext) {
+class ServiceRegistration(
+  serviceInfo:         ServiceInfo,
+  coordinatedShutdown: CoordinatedShutdown,
+  config:              Config,
+  registry:            ServiceRegistry
+)(implicit ec: ExecutionContext) {
```

Comment:
In case anyone else looks at this and has trouble spotting the difference (other than the formatting change): this removes the `ApplicationLifecycle` parameter and replaces it with `CoordinatedShutdown`.